api_tests/files/views/test_file_list.py

```diff
@@ -1,138 +1,159 @@
-        self.file1.add_tag('new', Auth(self.user))
-        self.file2.add_tag('new', Auth(self.user))
-        res = self.app.get(
+    def test_filter_on_single_tag(self, app, user, node, file_one, file_two, file_three, file_four):
+        file_one.add_tag('new', Auth(user))
+        file_two.add_tag('new', Auth(user))
+        file_three.add_tag('news', Auth(user))
+
+        # test_filter_on_tag
+        res = app.get(
             '/{}nodes/{}/files/osfstorage/?filter[tags]=new'.format(
-                API_BASE, self.node._id
+                API_BASE, node._id
             ),
-            auth=self.user.auth
+            auth=user.auth
         )
         data = res.json.get('data')
-        nt.assert_equal(len(data), 2)
+        assert len(data) == 2
         names = [f['attributes']['name'] for f in data]
-        nt.assert_in('file1', names)
-        nt.assert_in('file2', names)
-
-    def test_filtering_tags_exact(self):
-        self.file1.add_tag('cats', Auth(self.user))
-        self.file2.add_tag('cats', Auth(self.user))
-        self.file1.add_tag('cat', Auth(self.user))
-        res = self.app.get(
-            '/{}nodes/{}/files/osfstorage/?filter[tags]=cat'.format(
-                API_BASE, self.node._id
+        assert 'file_one' in names
+        assert 'file_two' in names
+
+        # test_filtering_tags_exact
+        res = app.get(
+            '/{}nodes/{}/files/osfstorage/?filter[tags]=news'.format(
+                API_BASE, node._id
             ),
-            auth=self.user.auth
+            auth=user.auth
         )
-        nt.assert_equal(len(res.json.get('data')), 1)
+        assert len(res.json.get('data')) == 1
 
-    def test_filtering_tags_capitalized_query(self):
-        self.file1.add_tag('cat', Auth(self.user))
-        res = self.app.get(
-            '/{}nodes/{}/files/osfstorage/?filter[tags]=CAT'.format(
-                API_BASE, self.node._id
+        # test_filtering_tags_capitalized_query
+        res = app.get(
+            '/{}nodes/{}/files/osfstorage/?filter[tags]=NEWS'.format(
+                API_BASE, node._id
             ),
-            auth=self.user.auth
+            auth=user.auth
         )
-        nt.assert_equal(len(res.json.get('data')), 1)
+        assert len(res.json.get('data')) == 1
 
-    def test_filtering_tags_capitalized_tag(self):
-        self.file1.add_tag('CAT', Auth(self.user))
-        res = self.app.get(
+        # test_filtering_tags_capitalized_tag
+        file_four.add_tag('CAT', Auth(user))
+        res = app.get(
             '/{}nodes/{}/files/osfstorage/?filter[tags]=cat'.format(
-                API_BASE, self.node._id
+                API_BASE, node._id
             ),
-            auth=self.user.auth
+            auth=user.auth
         )
-        nt.assert_equal(len(res.json.get('data')), 1)
+        assert len(res.json.get('data')) == 1
+
+    def test_filtering_on_multiple_tags(self, app, user, node, file_one, file_two, file_three, file_four):
```

Comment:
This looks like mutation, and I think you changed the test to account for the mutation. @alexschiller